File: RadiantSDK/TrueTest/Visual Studio 2013/VB.NET/Device Interface Standard/SimplePattern.vb
Imports TrueTestPatternGenerator
Imports System.Xml.Serialization

<Serializable()>
Public Class SimplePattern
    Inherits PatternBase

    <XmlIgnore()>
    Protected Overrides ReadOnly Property Name As String
        Get
            Return "Numeric Image #" & ImageNumber.ToString
        End Get
    End Property

#Region "pattern-specific properties"

    Public Property ImageNumber As Integer = 0

#End Region

End Class

File: RadiantSDK/TrueTest/Visual Studio 2013/C#/Device Interface Standard/SimplePattern.cs
using System;
using TrueTestPatternGenerator;
using System.Xml.Serialization;

namespace StandardDI
{
    [Serializable()]
    public class SimplePattern : PatternBase
    {
        [XmlIgnore()]
        protected override string Name
        {
            get
            {
                return "Numeric Image #" + ImageNumber.ToString();
            }
        }
        
        public int ImageNumber { get; set; }

        SimplePattern()
        {
            ImageNumber = 1;
        }
    }
}
